backend/dataall/modules/dataset_sharing/services/share_managers/s3_access_point_share_manager.py

```diff
@@ -541,6 +558,29 @@ def generate_default_kms_decrypt_policy_statement(target_requester_arn):
             "Resource": "*"
         }
 
+    @staticmethod
+    def generate_enable_pivot_role_permissions_policy_statement(dataset_account_id):
+        return {
+            "Sid": f"{DATAALL_ACCESS_POINT_ENABLE_PIVOT_ROLE_PERMISSIONS_SID}",
+            "Effect": "Allow",
+            "Principal": {
+                "AWS": [
+                    f"arn:aws:iam::{dataset_account_id}:role/dataallPivotRole"
```

Comment:
pivot role could also be `dataallPivotRole-cdk` - should look up from SSM what the pivotRoleName is